### PR TITLE
Test for DataFrame

### PR DESCRIPTION
**Pull request by hkchekc**
added test for DataFrame

### CODE REVIEW

**Comment by hkchekc** [comment]
Also some features in dataframe object is changed

File: nixio/data_frame.py [hunk 1]
```diff
@@ -101,40 +109,44 @@ def write_rows(self, rows, index=None):
             self._write_data(cr_list, sl=index)
 
     def read_rows(self, index):
+        if hasattr(index, '__iter__'):
```
Comment (achilleas-k) [inline nixio/data_frame.py]: It's best to check if it's an instance of Iterable than to check if it has an attribute.

In general though, I'm not sure of the purpose here. Is it for cases where the user supplies a list or tuple of indices? Would it work in that case?

**Comment by achilleas-k** [comment]
I dumped some thoughts about DataFrame creation in an issue here #353.

We probably don't need to support all these in this PR, but some basic way of creating a valid DF should be possible and I think currently it isn't very friendly.

**Comment by hkchekc** [comment]
The test now should be passing and most problems of py2 is resolved. The only thing is there is some strange problem with using a np array (with py2) to append rows. The types will be changed to some strange way  (to all strings when there is one string arguement) and forcing it back to the expected structured array will not help (there will be still a bad type arguement Error)

**Comment by achilleas-k** [comment]
> The test now should be passing and most problems of py2 is resolved. The only thing is there is some strange problem with using a np array (with py2) to append rows. The types will be changed to some strange way (to all strings when there is one string arguement) and forcing it back to the expected structured array will not help (there will be still a bad type arguement Error)

I've created an issue for this #356. Can you add a minimal example of the problem in a comment in that issue so we can look at it? I'm going to merge this PR but we should figure out what exactly is going on there.

File: nixio/data_frame.py [hunk 2]
```diff
@@ -31,6 +31,10 @@ def _create_new(cls, nixparent, h5parent, name, type_, shape, col_dict, compress
         return newentity
 
     def append_column(self, column, name, datatype=None):
+        if len(column) < len(self):
```
[the remark this answers is not in the document]
Comment (achilleas-k) [reply nixio/data_frame.py]: Was this resolved?

**Comment by achilleas-k** [comment]
Merging.

@hkchekc Make a not to simplify https://github.com/G-Node/nixpy/pull/352/files#r235896688 in the next PR. I don't want to keep this open just for that.

**Pull request merged**